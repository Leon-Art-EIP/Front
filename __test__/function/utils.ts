import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
