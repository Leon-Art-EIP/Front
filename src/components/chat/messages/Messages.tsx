import { useChat } from "../../../contexts/ChatContext";
import { ChatBox } from "./ChatBox";
import { ChatUserBanner } from "./ChatUserBanner";
import { MessageInput } from "./MessageInput";

export default function Messages(): JSX.Element {
  const { currentUser, currentChat, sendMessage } = useChat() || {};

  function getOtherUser() {
    if (currentChat?.UserOneId === currentUser?.user.id) {
      return currentChat?.UserTwoId;
    } else {
      return currentChat?.UserOneId;
    }
  }

  function handleSendMsg(msg: string) {
    var to = getOtherUser();
    if (to) {
      sendMessage(msg, to);
    }
  }

  return (
    <div className="flex flex-col h-full w-full">
      {currentChat && <ChatUserBanner currentChat={currentChat} currentUser={currentUser} />}
      <div className="flex flex-row w-full" style={{ height: "calc(100% - 6rem)" }}>
        <div className="flex flex-col h-full w-full">
          <ChatBox />
          <MessageInput handleSendMsg={handleSendMsg} />
        </div>
      </div>
    </div>
  );
}
