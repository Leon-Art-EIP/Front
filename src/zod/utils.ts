export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
