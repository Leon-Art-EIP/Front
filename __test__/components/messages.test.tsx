import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Messages from "../../src/components/chat/messages/Messages";

test("ChatBox snapshot", () => {
  testSnapshot(<Messages />);
});
