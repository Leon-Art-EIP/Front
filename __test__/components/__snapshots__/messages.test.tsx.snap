// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatBox snapshot 1`] = `
<div
  className="flex flex-col h-full w-full"
>
  <div
    className="flex flex-row w-full"
    style={
      {
        "height": "calc(100% - 6rem)",
      }
    }
  >
    <div
      className="flex flex-col h-full w-full"
    >
      <div
        className="flex flex-col h-full w-full overflow-y-auto gap-4"
        dir="btt"
      />
      <form
        className="flex justify-center items-end w-full gap-4 px-6 pb-6 pt-2 xl:px-10 xl:pb-6 xl:pt-2"
        onSubmit={[Function]}
      >
        <textarea
          className="text-tertiary w-full bg-background-inputfield outline-none placeholder:text-tertiary placeholder:opacity-50 resize-y overflow-y-auto rounded-3xl px-8 py-4"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Votre message..."
          style={
            {
              "height": "3.5rem",
              "maxHeight": "14rem",
              "minHeight": "3.5rem",
            }
          }
          value=""
        />
        <button
          className="bg-quaternary-hover p-3 rounded-full text-center"
          type="submit"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-7 w-7 fill-quaternary css-i4bv87-MuiSvgIcon-root"
            data-testid="SendIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
        </button>
      </form>
    </div>
  </div>
</div>
`;
